courseworks/week7/xuanzhaogao/hw7.jl

```diff
@@ -0,0 +1,1137 @@
+### A Pluto.jl notebook ###
+# v0.19.22
+
+using Markdown
+using InteractiveUtils
+
+# ╔═╡ 32ef4500-d88b-46c9-aa2c-b038768c4bd3
+using LinearAlgebra
+
+# ╔═╡ 54e1b7ea-70b2-44ea-aea4-40368b10f220
+using Plots
+
+# ╔═╡ f7248907-ac01-4ede-b0be-87fe7454794a
+using ForwardDiff
+
+# ╔═╡ 1ac79634-2483-4ed8-b474-66971b29273e
+md"""
+A unit-disk embedding of a graph is a way to represent the vertices and edges of a graph in the Euclidean plane such that each vertex is mapped to a distinct point and each edge is represented by a curve that connects the endpoints of the edge. In a unit-disk embedding, each vertex of the graph is represented by a disk of unit radius and the disks corresponding to adjacent vertices intersect if and only if the corresponding vertices are adjacent in the graph. The goal of a unit-disk embedding is to find a geometric representation of the graph that preserves its connectivity properties and is aesthetically pleasing. Unit-disk embeddings have applications in wireless sensor networks, where the nodes are represented by disks and the communication range between nodes is limited to the radius of their disks.
+"""
+
+# ╔═╡ 8f183ba7-fba6-4eea-b6d3-732e7cbe554a
+N = 10
+
+# ╔═╡ adc3c8b0-f385-461e-827d-5d7ce3e7c34c
+E = [(1, 2), (1, 3),
+	(2, 3), (2, 4), (2, 5), (2, 6),
+	(3, 5), (3, 6), (3, 7),
+	(4, 5), (4, 8),
+	(5, 6), (5, 8), (5, 9),
+	(6, 7), (6, 8), (6, 9),
+	(7,9), (8, 9), (8, 10), (9, 10)]
+
+# ╔═╡ 492e08b2-bcf9-4506-91d3-39620a9f9af3
+function distance!(x, i, j)
```

Comment:
```suggestion
function distance(x, i, j)
```

`!` is not needed if the function does not change inputs. Here `x` is just used for input.

Reply:
Thanks a lot, I removed these misused ```!```.